spec/std/socket_spec.cr

```diff
@@ -202,4 +231,51 @@ describe "UDPSocket" do
     client.close
     server.close
   end
+
+  it "sends and receives messages by sendto and recvfrom over IPv4" do
+    server = UDPSocket.new
+    server.bind("localhost", 12347)
```

Comment:
localhost is not guaranteed to resolve to a v4 address, use a `127.0.0.0/8` address explicitly like you did for v6.